### PR TITLE
Feature/scoped allocator adaptor

### PR DESCRIPTION
**Pull request by LukasKerk**

### CODE REVIEW

File: include/tsl/sparse_hash.h [hunk 1]
```diff
@@ -1082,9 +1091,9 @@ class sparse_hash : private Allocator,
   using sparse_buckets_allocator = typename std::allocator_traits<
       allocator_type>::template rebind_alloc<sparse_array>;
   using sparse_buckets_container =
-      std::vector<sparse_array, sparse_buckets_allocator>;
+      std::vector<sparse_array, sparse_buckets_allocator>;//here
```
Comment (bigerl) [inline include/tsl/sparse_hash.h]: left over comment?

File: include/tsl/sparse_hash.h [hunk 2]
```diff
@@ -1231,7 +1240,7 @@ class sparse_hash : private Allocator,
        * We can't use `vector(size_type count, const T& value, const Allocator&
        * alloc)` as it requires the value T to be copyable.
        */
-      m_sparse_buckets_data.resize(
+      m_sparse_buckets_data.resize( //TODO: this is the current critical point
```
Comment (bigerl) [inline include/tsl/sparse_hash.h]: left over comment?

File: include/tsl/sparse_hash.h [hunk 3]
```diff
@@ -1082,9 +1091,9 @@ class sparse_hash : private Allocator,
   using sparse_buckets_allocator = typename std::allocator_traits<
       allocator_type>::template rebind_alloc<sparse_array>;
   using sparse_buckets_container =
-      std::vector<sparse_array, sparse_buckets_allocator>;
+      std::vector<sparse_array, sparse_buckets_allocator>;//here
 
- public:
+public:
```
Comment (bigerl) [inline include/tsl/sparse_hash.h]: whitespace change

**Review by bigerl — CHANGES_REQUESTED** [review]
looks fine. only some formatting issues

File: include/tsl/sparse_hash.h (hunk 1 above)
Comment (LukasKerk) [reply include/tsl/sparse_hash.h]: You are completely correct. I removed it.

File: include/tsl/sparse_hash.h (hunk 2 above)
Comment (LukasKerk) [reply include/tsl/sparse_hash.h]: You are completely correct. I removed it.

File: include/tsl/sparse_hash.h (hunk 3 above)
Comment (LukasKerk) [reply include/tsl/sparse_hash.h]: fixxed

**Review by bigerl — APPROVED** [review]
👍🏻

**Pull request merged**